src/sentry/issues/endpoints/group_events.py

```diff
@@ -106,11 +108,22 @@ def _get_events_snuba(
             params["environment"] = [env.name for env in environments]
 
         full = request.GET.get("full") in ("1", "true")
+        random = request.GET.get("random") in ("1", "true")
+
+        if random:
+            orderby = ["random_number()"]
```

Comment:
To get a sample of data elsewhere, we've used https://github.com/getsentry/sentry/blob/c4ec651c5e58e54f70e25ba6d0ba4f1c755775bd/src/sentry/search/snuba/executors.py#L324

Not sure if it's meaningfully different though. I am slightly concerned that using `random_number()` here means we have to generate a random for every row and then scan on it, but I guess it's going to need to scan the entire time range to do anything like this.

I wonder if you could also just sort on `event_id`, since it's a uuid and is effectively random

Reply:
hmm... so this basically is "random" but deterministic and reproducible, as in same query with same parameters and same data will yield the same results. I guess that could be useful.

Reply:
I'm also not sure if it's more expensive to run `random_number()` on every row vs just having a pre-generated random we sort on.

Where did `random_number()` come from btw? Is it a snql thing? I don't see it in the clickhouse docs

Reply:
https://medium.com/datadenys/how-to-use-clickhouse-sampling-properly-db169faedfd7 This article mentions using `random()` is slow. I wonder if we should be figuring out how to use SAMPLE instead

Reply:
It's `rand()` in ClickHouse. Aliased as `random_number()` in our `discover.function_converter`

https://github.com/getsentry/sentry/blob/2fcdcff7edf9f1b324980e47fa4269b8f985584b/src/sentry/search/events/datasets/discover.py#L417-L426

Reply:
I'm inclined to agree with @wedamija here. What you could do is sort on the cityHash64 of the event id or occurrence id depending on which table it ends up using. A few advantages of this are
1. It _might_ be faster given that the tables contain `cityHash64(event_id)` and `cityHash64(occurrence_id)` in the ORDER BY. Worth checking out if these queries are slow.
2. The results are pseudo random and using the same limit + offset, you can get fetch consistent results. Which ties in to the next point which is the ability to paginate the results.
3. Also, snuba has a cache on the query, so for a period of time, while your query result is still cached, this query will just return the same results every time.

Reply:
OK, I'll close this one and open new PR with sample based on event_id (either directly or hash with some salt).